src/services/ReservationService.js

```diff
@@ -85,9 +88,38 @@ const hasReservations = async user => {
   return count > 0;
 };
 
+const claimReservation = async (reservationId, user) => {
+  if (!reservationId) {
+    throw new ReservationServiceError(
+      'Reservation cannot be claim without reservationId',
+    );
+  }
+  try {
+    const reservation = await checkReservation(reservationId);
+
+    // set reservation as claimed and create a connection between users
+    reservation.set('isClaimed', true);
+    reservation.set('user', user);
+    await reservation.save(null, { useMasterKey: true });
+    const fromUser = reservation.get('createdBy');
+    await ConnectionService.createConnection(fromUser, user);
+    const uniqueId = hat();
+    // create a channel between 2 users.
+    const channel = await ChatService.createChatChannel(
```

Comment:
Add a check to see if a channel already exists between the two users. If one doesnt exist, then create one.

Reply:
I think we could use a connection class and store a unique id channel. If it already exists a new one won't be created.